tools/harness-simulation/harness/Thread_Harness/Sniffer/SimSniffer.py

```diff
@@ -66,12 +152,40 @@ def __init__(self, **kwargs):
         self._local_pcapng_location = None
 
         if self.addr_port is not None:
+            # Replace `MsgBox` only once when simulation devices exist
+            if not SimSniffer.replaced_msgbox:
+                SimSniffer.replaced_msgbox = True
+                self._replaceMsgBox()
+
             self._sniffer = grpc.insecure_channel(self.addr_port)
             self._stub = sniffer_pb2_grpc.SnifferStub(self._sniffer)
 
             # Close the sniffer only when Harness exits
             win32api.SetConsoleCtrlHandler(self.__disconnect, True)
 
+    @watched
+    def _replaceMsgBox(self):
+        self.test_9_2_9_leader = RFEnclosureManager(['Router_1', 'Router_2'], ['DUT', 'Commissioner'], 1)
```

Comment:
```suggestion
        test_9_2_9_leader = RFEnclosureManager(['Router_1', 'Router_2'], ['DUT', 'Commissioner'], 1)
```
seems unnecessary to use member fields?

Reply:
I was afraid that local variables would deconstruct at some time, making their methods dangling references. Now I think I made a mistake. Python can handle such cases properly with reference counting.

Reply:
Yes, Python automatically collect garbages by reference counting, which is very efficient.

Reply:
In Python you rarely need to worry about dangling references.